plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -707,10 +621,12 @@ private void fillNextEntries()
 
             // process page
             for (int i = 0; i < extractors.size(); ++i) {
-                DeltaLakeTransactionLogEntry entry = extractors.get(i).getEntry(session, page.getBlock(i).getLoadedBlock(), pagePosition);
-                if (entry != null) {
-                    nextEntries.add(entry);
+                Block block = page.getBlock(i).getLoadedBlock();
+                if (block.isNull(pagePosition)) {
+                    continue;
                 }
+                DeltaLakeTransactionLogEntry entry = extractors.get(i).getEntry(session, (RowType) columnFields.get(i).getType(), block, pagePosition);
```

Comment:
This implementation has a bug when the requested field doesn't exist in Parquet file. Fixing the issue now.